File: java/mission-to-mars-solid/src/main/java/mars/MouvementSud2D.java
package mars;

public class MouvementSud2D implements Mouvement2D {

	public Position avancer(Position position) {
		return new Position(position.getX(), position.getY() - 1, position.getZ());
	}
	
	public Position reculer(Position position) {
		return new Position(position.getX(), position.getY() + 1, position.getZ());
	}
}

File: python/mission-to-mars-solid/src/mars/MouvementSud2D.py
import abc

from src.mars.Mouvement2D import Mouvement2D
from src.mars.Position import Position


class MouvementSud2D(Mouvement2D):

    def avancer(self, position: Position) -> Position:
        return Position(position.x, position.y - 1, position.z)

    def reculer(self, position: Position) -> Position:
        return Position(position.x, position.y + 1, position.z)
